source/runtime.cpp

```diff
@@ -243,9 +243,30 @@ void reshade::runtime::on_present()
 
 bool reshade::runtime::load_effect(const std::filesystem::path &path, size_t index)
 {
+	const std::string effect_name = path.filename().u8string();
+	const ini_file preset(_current_preset_path); // ini_file::load_cache is not thread-safe, so load from file here
+
 	effect &effect = _effects[index]; // Safe to access this multi-threaded, since this is the only call working on this effect
+	effect = {};
 	effect.source_file = path;
-	effect.compile_sucess = true;
+	effect.compiled = true;
+
+	if (!_effect_load_skipping || _load_option_disable_skipping)
+		effect.skipped = false;
+	else if (std::vector<std::string> techniques; !preset.get({}, "Techniques", techniques))
+		effect.skipped = false;
+	else if (std::find_if(techniques.cbegin(), techniques.cend(), [&effect_name](const std::string &technique) {
+		const size_t at_pos = technique.find('@') + 1;
+		return at_pos == 0 || technique.find(effect_name, at_pos) == at_pos; }) != techniques.cend())
+		effect.skipped = false;
+	else
+		effect.skipped = true;
+
+	if (effect.skipped)
```

Comment:
I think doing this check inside `load_effect` could be a problem when editing an effect in the editor, since this check would trigger if saving there too. But if I save an effect in the editor I would expect it to get loaded, even if the preset has it not active.
My suggestion would be to do this check in the loop in `load_effects` which calls `load_effect` instead. That way disabled effects still are not loaded by default, but will be loaded when any other part of the code (like the editor) calls `load_effect`.

Reply:
See https://github.com/crosire/reshade/pull/182/commits/1e7647dbb271a12067faa135ca652d1f7c5565c4